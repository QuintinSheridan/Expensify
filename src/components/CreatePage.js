import React from 'react';


const CreatePage = () => (
    <div>
        This is from my create component
    </div>
);

export default CreatePage;