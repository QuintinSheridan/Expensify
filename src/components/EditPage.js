import React from 'react';


const EditPage = (props) => {
    console.log(props.match.params)
    return (
        <div>
            This is from my edit component {props.match.params.id}
        </div>
        )
};

export default EditPage;