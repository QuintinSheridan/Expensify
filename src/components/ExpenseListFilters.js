import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filtersActions';

export class ExpenseListFilters extends React.Component {
    state = {
        calenderFocused: null
    };

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    };

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused}))
    };

    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    };

    onSortChange = (e)=> {
        console.log(e.target.value);
        if(e.target.value === "amount") {
            this.props.sortByAmount()
        }
        else if (e.target.value === "date") {
            this.props.sortByDate()
        }
    }

    render() {
        return (
            <div>
                <input 
                    type="text" 
                    value={this.props.filters.text} 
                    onChange={this.onTextChange}
                />

                <select 
                    onChange={this.onSortChange}
                    value={this.props.filters.sortBy}
                >
                    <option value="date" default>Date</option>
                    <option value="amount">Amount</option>
                </select>

                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                    showClearDates={true}
                />
            </div>
        )
    }

}

// startDateId={"dwjkhqkehwqjkeq"}
// endDateId={"cxzvcxbzbczxbz"}


// const ExpenseListFilters = (props) => (
//     <div>
//         <input type="text" value={props.filters.text} onChange={(e)=> {
//             console.log(e.target.value);
//             props.dispatch(setTextFilter(e.target.value));
//         }}/>

//         <select onChange={(e)=> {
//             console.log(e.target.value);
//             if(e.target.value === "amount") {
//                 props.dispatch(sortByAmount())
//             }
//             else if (e.target.value === "date") {
//                 props.dispatch(sortByDate())
//             }
//         }}>

//             <option value="date" default>Date</option>
//             <option value="amount">Amount</option>
//         </select>

//         <DateRangePicker
//                     startDate={this.props.filters.startDate}
//                     endDate={this.props.filters.endDate}
//                     onDatesChange={this.onDatesChange}
//                     focusedInput={this.state.calenderFocused}
//                     onFocusChange={this.onFocusChange}
//                     numberOfMonths={1}
//                     isOutsideRange={() => false}
//                     showClearDates={true}
//                 />
//     </div>
// );

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);

