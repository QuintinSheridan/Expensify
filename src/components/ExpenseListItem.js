// Export a stateless functional component

import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({dispatch, id, description, amount, createdAt }) => (
    <div>
    <Link to={`/edit/${id}`}>
        <h3> {description} </h3>
        <p> {amount} - {createdAt}</p>
    </Link>

    </div>
)

export default ExpenseListItem;