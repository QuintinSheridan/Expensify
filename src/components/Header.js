import React from 'react'
import {NavLink} from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to='/' exact={true} activeStyle={{
            fontWeight: "bold",
            color: "red"
          }}> Home </NavLink>
        <NavLink to='/create' activeStyle={{
            fontWeight: "bold",
            color: "red"
          }}> Create </NavLink>
        <NavLink to='/help' activeStyle={{
            fontWeight: "bold",
            color: "red"
          }}> Help </NavLink>
    </header>
);


export default Header;