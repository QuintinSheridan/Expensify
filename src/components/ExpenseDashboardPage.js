import React from 'react';
import ExpenseList from './ExpensesList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpenseDashboardPage = () => (
    <div>
        This is from my dashboard component
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;