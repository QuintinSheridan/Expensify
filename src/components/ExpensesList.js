import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListenItem';
import selectExpenses from '../selectors/expensesSelector';

const ExpenseList = (props) => (
    <div>
        <h1>ExpenseList</h1>
        {props.expenses.map((expense) => {
            return (
                <ExpenseListItem key={expense.id} id= {expense.id} {...expense} />
            )
        })}
    </div>
)

const mapStateToProps = (state) => (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters),
    };
}

export default connect(mapStateToProps)(ExpenseList);
