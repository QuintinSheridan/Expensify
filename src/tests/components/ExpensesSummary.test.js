import { EnzymeAdapter } from 'enzyme';
import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test('should render single expense summary', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={1495} />)
    expect(wrapper).toMatchSnapshot()
});

test('should render summary for multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={3250} />)
    expect(wrapper).toMatchSnapshot()
});

