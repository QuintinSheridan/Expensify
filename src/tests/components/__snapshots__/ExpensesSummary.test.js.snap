// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render single expense summary 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $1,495.00
     
  </h1>
</div>
`;

exports[`should render summary for multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totaling 
    $3,250.00
     
  </h1>
</div>
`;
